packages/flutter/lib/src/painting/text_style.dart

```diff
@@ -238,7 +245,8 @@ class TextStyle extends Diagnosticable {
     String fontFamily,
     String package,
   }) : fontFamily = package == null ? fontFamily : 'packages/$package/$fontFamily',
-       assert(inherit != null);
+       assert(inherit != null),
+       assert(identical(color, null) || identical(foreground, null), _kColorForegroundWarning);
```

Comment:
Please add a TODO pointing at that dart bug you commented on. That we have to do this trick with "identical" is silly.